src/language-server/definition-finder.ts

```diff
@@ -63,4 +67,46 @@ export default class DefinitionFinder extends Handler {
     }
     return feature.sourceRange;
   }
+
+  async getReferences(params: ReferenceParams): Promise<Location[]> {
+    const locations: Location[] = [];
+    if (params.context.includeDeclaration) {
+      const definition = await this.getDefinition(params);
+      if (definition) {
+        if (Array.isArray(definition)) {
+          locations.push(...definition);
+        } else {
+          locations.push(definition);
+        }
+      }
+    }
+    const localPath =
+        this.converter.getWorkspacePathToFile(params.textDocument);
+    const position = this.converter.convertPosition(params.position);
+    const analysis = await this.analyzer.analyzePackage();
+    const document = analysis.getDocument(localPath);
+    if (!(document instanceof Document)) {
+      return locations;
+    }
+    const location =
+        await this.featureFinder.getAstAtPosition(document, position);
+    if (!location) {
+      return locations;
```

Comment:
when can this happen? e.g. empty document?

Reply:
Or if the position of the cursor is something that we don't have a location kind for, like a string literal, or some text.